inc/classes/class-srm-redirect.php

```diff
@@ -136,6 +136,16 @@ public function maybe_redirect() {
 					if ( ( strrpos( $redirect_to, '*' ) === strlen( $redirect_to ) - 1 ) ) {
 						$redirect_to = rtrim( $redirect_to, '*' ) . ltrim( substr( $requested_path, strlen( $wildcard_base ) ), '/' );
 					}
+				} elseif (
+					// Handle the case of no trailing slashes on redirect from and redirect to, when
+					// requested URL contains the trailingslash.
+					! $matched_path
+					&& ( strlen ( untrailingslashit( $redirect_to ) )  === strlen( $redirect_to ) )
+					&& ( strlen ( untrailingslashit( $redirect_from ) )  === strlen( $redirect_from ) )
+					&& ( strrpos( $normalized_requested_path, '/' ) === strlen( $normalized_requested_path ) - 1 )
+				) {
+					$redirect_to = untrailingslashit( $redirect_to );
+					$matched_path = true;
```

Comment:
$matched_path should not be set to true here.  At this point, we have only checked that the lengths of the strings are the same and that there is a slash as the last character of the request path.

In my testing, this causes incorrect redirect matching.  To test this out, set up a redirect with no asterisk in the redirect from path and you will see that the logic here causes all request paths that end with a slash to match that.

Reply:
Good work @mslinnea.

Confirming that commenting out `srm_create_redirect( '/two/', $redirect_to );` (seemingly unrelated to the `/something-else/` request) makes the unit tests all pass, but adding it back in makes the test fail, therefore proving that my initial logic is flawed.

I'm looking again at the logic, but I'm open to suggestions.